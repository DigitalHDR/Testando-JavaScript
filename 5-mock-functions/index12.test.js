//* %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
//*     | Mock Functions | Funções de Simulação
//*     Nome das funções de simulação (Mock names)
//*
//*             Matchers personalizados
//*
//* Por fim, para tornar menos exigente afirmar como as
//* funções fictícias foram chamadas, adicionamos algumas
//* funções de correspondência personalizadas para você:

//* %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

test('custom matchers', () => {
  const mockFunc = jest.fn()

  const arg1 = 'arg1'
  const arg2 = 'arg2'

  mockFunc(arg1, arg2)

  const arg3 = 'arg3'
  const arg4 = 'arg4'

  mockFunc(arg3, arg4)

  // A função simulada foi chamada pelo menos uma vez
  expect(mockFunc).toBeCalled()

  // A função simulada foi chamada pelo menos uma vez com os argumentos especificados
  expect(mockFunc).toBeCalledWith(arg1, arg2)

  // A última chamada para a função simulada foi chamada com os argumentos especificados
  expect(mockFunc).lastCalledWith(arg3, arg4)

  // Todas as chamadas e o nome da simulação são gravados como um instantâneo
  expect(mockFunc).toMatchSnapshot()
})
