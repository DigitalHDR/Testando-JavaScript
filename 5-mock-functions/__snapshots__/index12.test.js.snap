// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom matchers 1`] = `
[MockFunction] {
  "calls": [
    [
      "arg1",
      "arg2",
    ],
    [
      "arg3",
      "arg4",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
